invenio_rdm_records/services/services.py

```diff
@@ -128,3 +128,26 @@ def oai_result_item(self, identity, oai_record_source):
             record,
             links_tpl=self.links_item_tpl,
         )
+
+    @unit_of_work()
+    def reindex_latest_first(self, identity, id_, extra_filter=None, uow=None):
+        """Reindexes records matching the query filter, prioritizing latest versions.
+
+        Records will be retrieved from search engine and current versions re-indexed right away.
+        Related records (siblings) will be reindexed afterwards.
+        """
+        self.require_permission(identity, "search")
+        params = {"q": extra_filter} if extra_filter else {}
+
+        records = self.scan(identity, params=params)
+
+        for record in records:
+            record, parent = self._get_record_and_parent_by_id(record["id"])
+            # Commit (and re-index) latest versions
+            uow.register(RecordCommitOp(parent))
+            if record:
+                uow.register(RecordCommitOp(record))
+
+            self._index_related_records(record, parent, uow=uow)
```

Comment:
This function might be a bottleneck since it will query DB directly and re-index one by one. 

Maybe we can use bulk indexing instead?